# Исследование надёжности заёмщиков

Во второй части проекта вы выполните шаги 3 и 4. Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом.

## Описание проекта
Заказчик — кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.<br>
Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

**Описание данных**
- `children` — количество детей в семье
- `days_employed` — общий трудовой стаж в днях
- `dob_years` — возраст клиента в годах
- `education` — уровень образования клиента
- `education_id` — идентификатор уровня образования
- `family_status` — семейное положение
- `family_status_id` — идентификатор семейного положения
- `gender` — пол клиента
- `income_type` — тип занятости
- `debt` — имел ли задолженность по возврату кредитов
- `total_income` — ежемесячный доход
- `purpose` — цель получения кредита

## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
# Сведём исследуемые данные в одну таблицу, где их сгруппируем по количеству детей и посчитаем число фактов задолженностей,
# выведя их долю от общего количества в данной категории.
tab = data.pivot_table(index = ['children'], values = 'debt', aggfunc = {'count','sum'})
tab = tab.rename(columns={'count':'total_credits','sum':'payment_delays'})
tab['proportion'] = round(tab['payment_delays']/tab['total_credits'], 3)
tab['proportion'] = tab['proportion'].transform(lambda x: '{:,.1%}'.format(x))
tab = tab.sort_values(by='proportion', ascending=True)
display(tab)

total_credits  payment_delays proportion
children                                          
5                     9               0       0.0%
0                 14091            1063       7.5%
3                   330              27       8.2%
1                  4808             444       9.2%
2                  2052             194       9.5%
4                    41               4       9.8%

**Вывод:** 
можно сказать, что видна _маленькая_ зависимость погашения кредита в срок от количества детей. Данные по заёмщикам с 5 детьми в анализе можно не учитывать из-за их малого количества (9), хотя у них и нет ни одной просрочки.<br>
_Больше всего_ кредитов у бездетных, однако у них же _меньше всего_ доля просрочек платежей. Такой факт можно объяснить их _более сильной платёжеспособностью_ из-за меньших расходов.<br>
Также _мало просрочек_ у заёмщиков с 3 детьми, видимо, из-за большей их ответственности (сюда можно отнести семьи с 5 детьми). Однако из числа _ответственных_ многодетных заёмщиков _выпадает_ группа с 4 детьми, доля которых наибольшая в числе просрочек - почти 10%. В то же время их количество _непоказательно_ для того, чтобы делать однозначный вывод, потому то общее число многодетных заёмщиков (3-5 детей) 380 человек, что составляет менее 1,8% от общего числа (21525).<br>
Одновременно у людей с 1-2 детьми _практически одинаковые_ доли в просрочках по кредитам, но у семей с 2 детьми доля просрочек больше. Это можно объяснить более высокой финансовой нагрузкой.<br>
Для _полного подтверждения гипотезы_ нужно больше данных по многодетным семьям, хотя можно предположить, что у них _меньше необходимости_ прибегать к кредитным продуктам из-за детских пособий, которые составляют 40% от средней зарплаты за предыдущие два года или от 50 до 100% прожиточного минимума в регионе. Ещё здесь роль играет материнский капитал.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [23]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
tab_family = data.pivot_table(index = [ 'family_status'], values = 'debt', aggfunc = {'count','sum'})
tab_family = tab_family.rename(columns={'count':'total_credits','sum':'payment_delays'})
tab_family['proportion'] = round(tab_family['payment_delays']/tab_family['total_credits'], 3)
tab_family['proportion'] = tab_family['proportion'].transform(lambda x: '{:,.1%}'.format(x))
tab_family = tab_family.sort_values(by='proportion', ascending=True)
display(tab_family)

total_credits  payment_delays proportion
family_status                                                  
вдовец / вдова                   951              63       6.6%
в разводе                       1189              84       7.1%
женат / замужем                12261             927       7.6%
гражданский брак                4134             385       9.3%
Не женат / не замужем           2796             273       9.8%

**Вывод:** 
Здесь видим, что _меньше всего_ люди в разводе (чуть более 7%) и вдовцы (6,6%) допускают просрочки возврата кредита. Это можно объяснить сниженными расходами и более высоким уровнем доходов, опытом и возрастом, повышенной ответственностью из-за необходимости рассчитывать только на себя. При этом их количество достаточное для релевантной выборки.<br>
Люди, состоящие в официальном браке, на которых приходится _больше всего кредитов_, допускают просрочки почти в 8% случаев. Это можно объяснить повышенными расходами из-за наличия детей и финансовыми трудностями из-за того, что в семье работает только один человек.<br>
_Больше всего_ доли просрочек у заёмщиков в гражданском браке (более 9%) и неженатых (почти 10%), что можно объяснить молодым возрастом, низким уровнем доходов, слабой ответственностью из-за недостаточного опыта, наличием детей и пр.
В общем, можно сказать, что зависимость прослеживается ясно.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [24]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
tab_income = data.pivot_table(index = [ 'total_income_category'], values = 'debt', aggfunc = {'count','sum'})
tab_income = tab_income.rename(columns={'count':'total_credits','sum':'payment_delays'})
tab_income['proportion'] = round(tab_income['payment_delays']/tab_income['total_credits'], 3)
tab_income['proportion'] = tab_income['proportion'].transform(lambda x: '{:,.1%}'.format(x))
tab_income = tab_income.sort_values(by='proportion', ascending=True)
display(tab_income)

total_credits  payment_delays proportion
total_income_category                                          
D                                349              21       6.0%
B                               5014             354       7.1%
A                                 25               2       8.0%
C                              15921            1353       8.5%
E                                 22               2       9.1%

**Вывод:** 
напомним, что категории по доходу у нас следующие:<br>
A — 1 млн руб. и выше;<br>
B — 200 тыс. до 1 млн руб.;<br>
C — 50–200 тыс. руб.;<br>
D — 30–50 тыс. руб;<br>
E — 0–30 тыс. руб.<br>
Доля невозвратов _меньше всего_ (6%) у заёмщиков из категории D, то есть с доходом от 30 до 50 тысяч рублей. Скорее всего, они стараются избегать штрафов за просрочку, потому что доходы не позволяют их оплачивать. Одновременно на них приходится _менее 8% от всех выданных кредитов_.<br>
Также _меньше всего_ (0,1%) полученных кредитов у людей с доходами до 30 тыс. рублей (категория Е), потому что банки _крайне неохотно_ выдают им займы. При этом у них _самая дольшая_ доля просрочек — более 9%, что можно также объяснить низким уровнем доходов или их непостоянством.<br>
В тоже время заёмщиков из категории А с доходами свыше 1 млн рублей также менее 0,1% от их общего числа, что говорит о _редкой необходимости в кредитовании_ из-за высокого уровня доходов. Однако в этой категории доля просрочек составляет 8%, что можно объяснить их _слабой ответственностью_, потому что доходы очень высокие.<br>
_Больше всего_ (почти 74%) получили кредитов заёмщики из категории С (доходы от 50 до 200 тыс. рублей), но у них же _довольно большая доля просрочек_ — 8,5%. С большой долей вероятности у них есть семьи, а может и не один кредит. Но здесь нужно провести более тонкую категоризацию и _разделить данную группу на несколько_, потому что доходы 50-80 тыс. рублей можно считать _слабыми_, а 150-200 тыс. рублей — _хорошими_. Такой шаг может дать более интересные результаты.<br>
Заёмщики с доходом от 200 тыс. до 1 млн рублей (категория В) на _втором месте_ по объёму кредитов (более 23%) и низкой доле просрочек (чуть выше 7%). Скорее всего, это предприниматели, о чём говорят высокие доходы и повышенная ответственность.<br>
При этом, сравнивая две самые многочисленные выборки B и C — если говорить кратко, то чем выше доходы, тем меньше просрочек.<br>
Здесь видно, что категория С самая большая, как по количеству кредитов, так и по диапазону доходов. То есть категория может включать в себя разные группы заёмщиков: с небольшим доходом в 50-70 тыс. рублей или 80-100 тыс. рублей, что считается уровнем дохода ниже среднего. В эту же категорию входят заёмщики с достаточно высоким уровнем 150-200 тыс. рублей. Поэтому желательно ещё категоризировать эту группу, чтобы увидеть больше информации.<br>
Однако, именно из-за такого большого разброса в уровнях дохода у этой категории больше доля просрочек, что опять же можно объяснить недостаточностью финансов, у заёмщиков ближе к нижней границе диапазона.<br>
Та же ситуация наблюдается в категории В, потому что разброс цифр ещё больше. Тем более, что деятельность предпринимателей всегда связана с определённым риском, что напрямую влияет на появление просрочек по кредитам.<br>
Итак, мы видим, что наша гепотеза о зависимости возврата кредита в срок от уровня дохода имеет подтверждение.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [25]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
tab_goals = data.pivot_table(index = [ 'purpose_category'], values = 'debt', aggfunc = {'count','sum'})
tab_goals = tab_goals.rename(columns={'count':'total_credits','sum':'payment_delays'})
tab_goals['proportion'] = round(tab_goals['payment_delays']/tab_goals['total_credits'], 3)
tab_goals['proportion'] = tab_goals['proportion'].transform(lambda x: '{:,.1%}'.format(x))
tab_goals = tab_goals.sort_values(by='proportion', ascending=True)
display(tab_goals)

total_credits  payment_delays proportion
purpose_category                                                  
операции с недвижимостью          10751             780       7.3%
проведение свадьбы                 2313             183       7.9%
операции с автомобилем             4279             400       9.3%
получение образования              3988             369       9.3%

**Вывод:** 
мы видим прямую взаимосвязь. _Самая низкая_ доля просрочек (7,3%) у заёмщиков, которые стремятся улучшить свои жилищные условия. На них же приходится и _самая большая_ доля выданных кредитов — почти 50%. Это можно объяснить их высокой ответственностью, потому что жильё является _сильным фактором_ для планирования расходов и повышения доходов.<br>
На втором месте _по надёжности_ (менее 8% просрочек) заёмщики, которые планируют свадьбу, хотя на них приходится _менее 11% выданных займов_. Этот результат также можно объяснить чувством ответственности, которое растёт после принятия решения о _создании семьи_. Ещё сюда можно отнести _финансовую помощь_, которую часто оказывают родственники молодым семьям.<br>
_Самые высокие_ доли просрочек — по 9,3% — среди заёмщиков, которые _планировали операции с автомобилями_ (покупка, ремонт и т.п.) и _получение образования_ (оплата учёбы). В тоже время, на них приходится _почти 20%_ и _18,5%_ соответственно от общего объёма выданных кредитов. В этих случаях, вероятно, играет роль _сумма кредита_, потому что при покупке новой машины людям свойственно _завышать запросы_ и _переоценивать свои силы_ (хорошо бы разделить покупку и ремонт, чтобы более чётко увидеть уровень надёжности заёмщиков).<br>
В случае с учёбой _высокую долю_ невозвратов можно объяснить _возрастом_ и _низкой ответственностью_, потому что за образованием, в основном, идут молодые люди, у которых ещё мало опыта.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:*<br>
пропуски данных наблюдаются в столбцах days_employed (_общий трудовой стаж в днях_) и total_income (_ежемесячный доход_). Это можно объяснить человеческим фактором: возможно, люди не захотели вносить эти данные или посчитали их слишком маленькими.<br>
Кроме того, в пропусках нужно отметить их _одинаковое количество в разных столбцах_, что может быть во время копирования, считывания или записи данных, изменения формата файла.<br>
Из этого напрашивается вывод, что факт _невнесения данных_ в один из столбцов _влияет_ на пропуски в другом.<br>

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:*<br>
столбцы с пропусками _days_employed_ и _total_income_ содержат количественные данные об общем времени трудового стажа и ежемесячном доходе, в которых _могут быть_ аномально высокие или низкие числа. Значит _среднее арифметическое_ здесь не подойдёт, потому что _исказит результат_. Лучше всего покажут выборку в целом с незаполненными данными медианные значения.<br>
Кроме того, в пропусках нужно отметить их _одинаковое количество в разных столбцах_. Из этого напрашивается вывод, что факт _невнесения данных_ в один из столбцов _влияет_ на пропуски в другом. Это _можно_ списать на технические ошибки, но _изначальной причиной_ всё равно остаётся человек.<br>

### Шаг 4: общий вывод.

В работу от кредитного отдела банка получены данные статистики о платёжеспособности клиентов. Была поставлена задача разобраться, влияет ли семейное положение и количество детей клиента на погашение кредита в срок и просрочки. Результаты исследования лягут в модель кредитного скоринга для оценки способности потенциального заёмщика вовремя вернуть кредит банку.<br>
На стадии предобработки были выявлены пропущенные значения в столбцах "общий трудовой стаж в днях" и "ежемесячный доход". Скорее всего, пропуски связаны с человеческим фактором. Поэтому заказчику нужно сделать данные поля обязательными для заполнения. Пропущенные значения были заменены на медианные показатели, как самые подходящие для получения релевантного результата. Также были обработаны дубликаты данных.<br>
В ходе исследования были получены выборки для изучения зависимости количества просрочек от семейного положения, количества детей, целей получения кредита и уровня доходов. Здесь нужно сказать, что самыми надёжными заёмщиками являются разведённые или овдовевшие лица без детей с невысоким постоянным доходом, потому что их финансовое положение не позволяет оплачивать штрафные санкции. В эту же группу входят люди, скорее всего предприниматели, с доходом выше 200 тыс. рублей в месяц, потому что бизнес требует ответственности и финансовой дисциплины, несмотря на все риски.<br>
Больше всего вероятность просрочки по кредиту у заёмщиков c невысоким уровнем доходов, которые неженаты или состоят в гражданском браке, имеют трёх и более детей, а целью кредита указывают учёбу или покупка автомобиля, или покупку недорогих товаров. При этом не обязательна полная комбинация всех этих факторов, даже некоторые из них позволяют говорить о росте вероятности просрочек платежей по кредиту в будущем.<br>
В тоже время зависимость данных от выдвинутых гипотез очень слабая, не прослеживается чёткой взаимосвязи. Однако проведённый анализ показывает, что информация при тщательном сборе и глубокой проработке может дать более точные результаты. Для этого нужно наладить получение информации без пропусков и дублей, ограничить определённым списком цели кредита, уровень доходов, наличие детей, чтобы не было аномалий, например минус 1.<br>
Для более точного прогнозирования, всё-таки желательно больше данных, например, добавить сведения об имуществе в собственности, обязательных расходах (съёмное жильё, к примеру) и т.п.